ci/do_ci.sh

```diff
@@ -85,6 +84,23 @@ else
   BUILD_ARCH_DIR="/linux/${ENVOY_BUILD_ARCH}"
 fi
 
+setup_clang_toolchain() {
+    CONFIG_PARTS=()
+    if [[ -n "${ENVOY_RBE}" ]]; then
+        CONFIG_PARTS+=("remote")
+    fi
+    CONFIG_PARTS+=("clang")
+    ENVOY_STDLIB="${ENVOY_STDLIB:-libc++}"
+    if [[ "${ENVOY_STDLIB}" == "libc++" ]]; then
+        CONFIG_PARTS+=("libc++")
+    fi
+    CONFIG="$(IFS=- ; echo "${CONFIG_PARTS[*]}")"
```

Comment:
Just my personal preference, it may be better to have multiple `--config` entries (e.g., `--conifg=remote --config=clang`) rather than one where the string is concatenated (and depends on the order of things).
As this is just my personal preference, it doesn't block this PR.

Reply:
yeah - i was trying to keep existing and just tidy up how it was done - cleaning up the bazel flags is an ongoing thing so ill consider this in future iterations